slides/aml-12-022618-model-evaluation/index.html

```diff
@@ -0,0 +1,695 @@
+<br />
+(i.e. a false positive costs me 10 dollars; a false negative, 100 dollars)
+- What guarantees do we want to give?
+
+???
+---
+# Changing Thresholds
+.tiny-code[
+```python
+data = load_breast_cancer()
+
+X_train, X_test, y_train, y_test = train_test_split(
+    data.data, data.target, stratify=data.target, random_state=0)
+
+lr = LogisticRegression().fit(X_train, y_train)
+y_pred = lr.predict(X_test)
+
+print(classification_report(y_test, y_pred))
+```
+```
+          precision   recall  f1-score  support
+0              0.91     0.92      0.92       53  
+1              0.96     0.94      0.95       90
+avg/total      0.94     0.94      0.94      143
+```
+```python
+y_pred = lr.predict_proba(X_test)[:, 1] > .85
+
+print(classification_report(y_test, y_pred))
+```
+```
+          precision   recall  f1-score  support
+0              0.84     1.00      0.91       53  
+1              1.00     0.89      0.94       90
+avg/total      0.94     0.93      0.93      143
+
+```
+]
+
+
+???
+---
+# Precision-Recall curve
+.smaller[
+```python
+X, y = make_blobs(n_samples=5000, centers=2, cluster_std=[7.0, 2],
+                  random_state=22, shuffle=False)
+# make classes imbalanced
+X, y = X[:3000], y[:3000]
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)
+
+svc = SVC(gamma=.05).fit(X_train, y_train)
+
+precision, recall, thresholds = precision_recall_curve(
+    y_test, svc.decision_function(X_test))
+# find threshold closest to zero:
+close_zero = np.argmin(np.abs(thresholds))
+plt.plot(precision[close_zero], recall[close_zero], 'o', markersize=10,
+         label="threshold zero", fillstyle="none", c='k', mew=2)
+```
+]
+
+???
+---
+# Precision-Recall curve
+.center[
+![:scale 75%](images/precision_recall_curve.png)
+]
+???
+---
+# Comparing RF and SVC
+.center[
+![:scale 75%](images/rf_vs_svc.png)
+]
+???
+
+---
+# Average Precision
+.center[
+![:scale 75%](images/avg_precision.png)
+]
+Same as area under the precision-recall curve
+<br />
+(depending on how you treat edge-cases)
+???
+---
+# F1 vs average Precision
+.tiny-code[
+```python
+from sklearn.metrics import f1_score
+print("f1_score of random forest: {:.3f}".format(
+      f1_score(y_test, rf.predict(X_test))))
+print("f1_score of svc: {:.3f}".format(f1_score(y_test, svc.predict(X_test))))
+```
+
+```
+f1_score of random forest: 0.709
+f1_score of svc: 0.715
+```
+```python
+from sklearn.metrics import average_precision_score
+ap_rf = average_precision_score(y_test, rf.predict_proba(X_test)[:, 1])
+ap_svc = average_precision_score(y_test, svc.decision_function(X_test))
+print("Average precision of random forest: {:.3f}".format(ap_rf))
+print("Average precision of svc: {:.3f}".format(ap_svc))
+```
+```
+Average precision of random forest: 0.682
+Average precision of svc: 0.693
+```
+]
+AP only considers ranking!
+???
+---
+
+# ROC Curve
+.center[
+![:scale 100%](images/zoo.png)
+]
+`$$ FPR = \frac{FP}{FP+TN}$$`
```

Comment:
Can you use ``\text`` for all the acronyms and also make the font bigger for this?

Reply:
Done